iothub_client/samples/pnp/pnp_temperature_controller/pnp_thermostat_component.c

```diff
@@ -0,0 +1,367 @@
+    int numTemperatureUpdates;
+    // Total of all temperature updates during current exceution run.  Used to determine average temperature of this thermostat component
+    double allTemperatures;
+}
+PNP_THERMOSTAT_COMPONENT;
+
+//
+// BuildUtcTimeFromCurrentTime writes the current time, in ISO 8601 format, into the specified buffer
+//
+static bool BuildUtcTimeFromCurrentTime(char* utcTimeBuffer, size_t utcTimeBufferSize)
+{
+    bool result;
+    time_t currentTime;
+    struct tm * currentTimeTm;
+
+    time(&currentTime);
+    currentTimeTm = gmtime(&currentTime);
+
+    if (snprintf(utcTimeBuffer, utcTimeBufferSize, g_ISO8601Format, currentTimeTm->tm_year + 1900, currentTimeTm->tm_mon, currentTimeTm->tm_mday, 
+                  currentTimeTm->tm_hour, currentTimeTm->tm_min, currentTimeTm->tm_sec) < 0)
+    {
+        LogError("snprintf on UTC time failed");
+        result = false;
+    }
+    else
+    {
+        result = true;
+    }
+
+    return result;
+}
+
+
+PNP_THERMOSTAT_COMPONENT_HANDLE PnP_ThermostatComponent_CreateHandle(const char* componentName)
+{
+    PNP_THERMOSTAT_COMPONENT* thermostatComponent;
+
+    if (strlen(componentName) > PNP_MAXIMUM_COMPONENT_LENGTH)
+    {
+        LogError("componentName=%s is too long.  Maximum length is=%d", componentName, PNP_MAXIMUM_COMPONENT_LENGTH);
+        thermostatComponent = NULL;
+    }
+    // On initial invocation, store the UTC time into g_programStartTime global.
+    else if ((g_programStartTime[0] == 0) && (BuildUtcTimeFromCurrentTime(g_programStartTime, sizeof(g_programStartTime)) == false))
+    {
+        LogError("Unable to store program start time");
+        thermostatComponent = NULL;
+    }
+    else if ((thermostatComponent = (PNP_THERMOSTAT_COMPONENT*)calloc(1, sizeof(PNP_THERMOSTAT_COMPONENT))) == NULL)
+    {
+        LogError("Unable to allocate thermostat");
+    }
+    else
+    {
+        strcpy(thermostatComponent->componentName, componentName);
+        thermostatComponent->currentTemperature = DEFAULT_TEMPERATURE_VALUE;
+        thermostatComponent->maxTemperature = DEFAULT_TEMPERATURE_VALUE;
+        thermostatComponent->minTemperature = DEFAULT_TEMPERATURE_VALUE;
+        thermostatComponent->numTemperatureUpdates = 1;
+        thermostatComponent->allTemperatures = DEFAULT_TEMPERATURE_VALUE;
+    }
+
+    return (PNP_THERMOSTAT_COMPONENT_HANDLE)thermostatComponent;
+}
+
+void PnP_ThermostatComponent_Destroy(PNP_THERMOSTAT_COMPONENT_HANDLE pnpThermostatComponentHandle)
+{
+    if (pnpThermostatComponentHandle != NULL)
+    {
+        free(pnpThermostatComponentHandle);
+    }
+}
+
+//
+// BuildMaxMinCommandResponse builds the response to the command for getMaxMinReport
+//
+static bool BuildMaxMinCommandResponse(PNP_THERMOSTAT_COMPONENT* pnpThermostatComponent, unsigned char** response, size_t* responseSize)
+{
+    int responseBuilderSize = 0;
+    unsigned char* responseBuilder = NULL;
+    bool result;
+    char currentTime[128];
```

Comment:
I think there a few time buffers used throughout. Should the size be `#define`'d?

Reply:
fixing.